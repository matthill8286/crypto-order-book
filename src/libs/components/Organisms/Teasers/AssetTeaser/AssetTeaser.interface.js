
//# sourceMappingURL=AssetTeaser.interface.js.map