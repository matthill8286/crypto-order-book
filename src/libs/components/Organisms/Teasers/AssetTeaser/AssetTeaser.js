function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import React from 'react';
import { useInView } from 'react-intersection-observer';
import { NativeLazyLoadOptions, Picture } from "../../../Atoms/Picture";
import { Tag } from "../../../Atoms/Tag";
import { CopyText, Heading, HeadingFeatured } from "../../../Atoms/Typography";
import { media, ALTERNATE, styled, ThemeConsumer } from "../../../../styles";
import { spacing } from "../../../../styles/sc-vars-global";
import { StyledBackgroundImage, StyledCard, StyledBadgeWrapper, StyledFlexBottomContainer, StyledInfoContainer, StyledInfoContainerContent, StyledInfoTextContainer, StyledLink, StyledMainContainer, StyledPictureContainer, StyledAssetContainer, StyledRect, StyledTongue, StyledTopTeaserTagWrapper, StyledUpsetRect, StyledWrapper } from "./AssetTeaser.styled";
import { Badge } from "../../../Atoms/Badge";
var StyledFFSaturn = styled.div.withConfig({
  displayName: "AssetTeaser__StyledFFSaturn",
  componentId: "a56of2-0"
})(["display:none;", ""], function (_ref) {
  var theme = _ref.theme;
  return theme.name === ALTERNATE && "\n  ".concat(media.md, " {\n    display: block;\n  }\n\n  > svg {\n    position: absolute;\n    left: -1px;\n    top: 0;\n    z-index: 1;\n    height: 100%;\n    width: auto;\n\n    ").concat(media.ie11, " {\n      width: 100px;\n      top: -7px;\n    }\n  }");
});

var Info = function Info(_ref2) {
  var contentText = _ref2.contentText,
      tags = _ref2.tags,
      limitLines = _ref2.limitLines;
  return React.createElement(StyledInfoTextContainer, null, React.createElement(CopyText, {
    fontSize: "sm",
    lineHeight: "sm",
    limitLines: limitLines ? limitLines : tags && tags.length > 0 ? 3 : 5,
    tag: "span"
  }, contentText));
};

var Tags = function Tags(_ref3) {
  var tags = _ref3.tags;

  if (tags && tags.length > 0) {
    return React.createElement(StyledFlexBottomContainer, null, React.createElement(StyledTopTeaserTagWrapper, {
      limitLines: 2
    }, tags.map(function (t, index) {
      return t && t.trim().length > 0 ? React.createElement(Tag, {
        key: 'Tag_' + t + index,
        text: t,
        marginBottom: true,
        marginRight: true
      }) : undefined;
    })));
  } else {
    return null;
  }
};

var Background = function Background(_ref4) {
  var backgroundVariant = _ref4.backgroundVariant,
      image = _ref4.image,
      imageDesktop = _ref4.imageDesktop,
      lazyloadBackgroundImage = _ref4.lazyloadBackgroundImage,
      hasAsset = _ref4.hasAsset,
      theme = _ref4.theme;

  var _useInView = useInView({
    triggerOnce: true
  }),
      _useInView2 = _slicedToArray(_useInView, 2),
      ref = _useInView2[0],
      inView = _useInView2[1];

  var hasQueryParams = (image === null || image === void 0 ? void 0 : image.indexOf('?')) !== -1;
  var imageUrl = imageDesktop ? imageDesktop : image;
  var lazyLoadImage = imageUrl ? "".concat(imageUrl).concat(hasQueryParams ? '&' : '?', "q=1") : undefined;

  if (theme === ALTERNATE) {
    return React.createElement(StyledRect, {
      ref: ref,
      image: imageUrl,
      inView: inView,
      lazyLoadImage: lazyLoadImage,
      hasAsset: hasAsset,
      lazyloadBackgroundImage: lazyloadBackgroundImage
    });
  } else {
    switch (backgroundVariant) {
      case 'tongue':
        return React.createElement(StyledTongue, {
          ref: ref,
          height: 140
        }, React.createElement(StyledBackgroundImage, {
          image: imageUrl,
          inView: inView,
          lazyLoadImage: lazyLoadImage,
          hasAsset: hasAsset,
          lazyloadBackgroundImage: lazyloadBackgroundImage
        }));

      case 'upsetRect':
        return React.createElement(StyledUpsetRect, {
          ref: ref,
          image: imageUrl,
          hasAsset: hasAsset
        });

      case 'rect':
        return React.createElement(StyledRect, {
          ref: ref,
          image: imageUrl,
          inView: inView,
          lazyLoadImage: lazyLoadImage,
          hasAsset: hasAsset,
          lazyloadBackgroundImage: lazyloadBackgroundImage
        });
    }
  }
};

var MainHeading = function MainHeading(_ref5) {
  var headline = _ref5.headline,
      themeName = _ref5.themeName,
      limitLines = _ref5.limitLines;
  return React.createElement(HeadingFeatured, {
    fixedMdSize: themeName === ALTERNATE,
    limitLines: limitLines
  }, headline);
};

export var AssetTeaser = function AssetTeaser(_ref6) {
  var smallHeadline = _ref6.smallHeadline,
      _ref6$background = _ref6.background,
      background = _ref6$background === void 0 ? 'grey' : _ref6$background,
      headline = _ref6.headline,
      contentText = _ref6.contentText,
      assetImage = _ref6.assetImage,
      backgroundImage = _ref6.backgroundImage,
      backgroundImageDesktop = _ref6.backgroundImageDesktop,
      lazyload = _ref6.lazyload,
      lazyloadBackgroundImage = _ref6.lazyloadBackgroundImage,
      tags = _ref6.tags,
      limitLines = _ref6.limitLines,
      _ref6$backgroundVaria = _ref6.backgroundVariant,
      backgroundVariant = _ref6$backgroundVaria === void 0 ? 'upsetRect' : _ref6$backgroundVaria,
      _ref6$elevation = _ref6.elevation,
      elevation = _ref6$elevation === void 0 ? 0 : _ref6$elevation,
      elevationHover = _ref6.elevationHover,
      link = _ref6.link,
      _ref6$fixedImageHeigh = _ref6.fixedImageHeight,
      fixedImageHeight = _ref6$fixedImageHeigh === void 0 ? false : _ref6$fixedImageHeigh,
      onClick = _ref6.onClick,
      _ref6$clickableWithou = _ref6.clickableWithoutLink,
      clickableWithoutLink = _ref6$clickableWithou === void 0 ? true : _ref6$clickableWithou,
      badges = _ref6.badges,
      rest = _objectWithoutProperties(_ref6, ["smallHeadline", "background", "headline", "contentText", "assetImage", "backgroundImage", "backgroundImageDesktop", "lazyload", "lazyloadBackgroundImage", "tags", "limitLines", "backgroundVariant", "elevation", "elevationHover", "link", "fixedImageHeight", "onClick", "clickableWithoutLink", "badges"]);

  var isClickable = !!onClick && clickableWithoutLink || !!(link !== null && link !== void 0 && link.href);
  return React.createElement(ThemeConsumer, null, function (theme) {
    return React.createElement(StyledLink, {
      to: link === null || link === void 0 ? void 0 : link.to,
      href: link === null || link === void 0 ? void 0 : link.href,
      onClick: onClick,
      underline: false
    }, React.createElement(StyledWrapper, null, React.createElement(StyledCard, _extends({
      elevation: elevation,
      elevationHover: elevationHover,
      display: "flex",
      fullHeight: true,
      noBorder: 'none',
      surface: background,
      badges: []
    }, rest), React.createElement(StyledMainContainer, {
      isClickable: isClickable
    }, React.createElement(StyledInfoContainer, {
      backgroundVariant: backgroundVariant,
      hasTags: !!tags && tags.length > 0,
      hasAsset: !!assetImage
    }, badges && badges.length > 0 && React.createElement(StyledBadgeWrapper, null, React.createElement(Badge, {
      actionType: "secondary",
      badges: badges
    })), React.createElement(StyledInfoContainerContent, {
      hasBadges: Boolean(badges && badges.length > 0)
    }, smallHeadline && React.createElement(Heading, {
      tag: "p",
      scale: "level-5",
      weight: "semibold",
      lineHeight: "sm",
      limitLines: 1,
      limitWidth: (theme === null || theme === void 0 ? void 0 : theme.name) === ALTERNATE ? '300px' : '350px',
      margin: "0 0 ".concat(spacing.base.xs, " 0"),
      uppercase: true
    }, smallHeadline), React.createElement(MainHeading, {
      headline: headline,
      themeName: theme === null || theme === void 0 ? void 0 : theme.name,
      limitLines: 2
    }), React.createElement(Info, {
      contentText: contentText,
      tags: tags,
      limitLines: limitLines
    }), React.createElement(Tags, {
      tags: tags
    }))), React.createElement(StyledAssetContainer, {
      fixedImageHeight: fixedImageHeight
    }, React.createElement(Background, {
      image: backgroundImage,
      imageDesktop: backgroundImageDesktop,
      lazyloadBackgroundImage: lazyloadBackgroundImage,
      backgroundVariant: backgroundVariant,
      hasAsset: !!assetImage,
      theme: theme === null || theme === void 0 ? void 0 : theme.name
    }), assetImage && React.createElement(StyledPictureContainer, null, React.createElement(Picture, {
      src: assetImage,
      height: "100%",
      loading: lazyload ? NativeLazyLoadOptions.Lazy : NativeLazyLoadOptions.Eager
    })))))));
  });
};
//# sourceMappingURL=AssetTeaser.js.map