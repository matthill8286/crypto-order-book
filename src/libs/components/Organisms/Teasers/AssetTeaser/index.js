export * from "./AssetTeaser";
export * from "./AssetTeaser.interface";
//# sourceMappingURL=index.js.map