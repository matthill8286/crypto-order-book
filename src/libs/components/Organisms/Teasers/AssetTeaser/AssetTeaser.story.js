function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { array, radios, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { StorybookRouterWrapper } from "../../../../utils/StorybookWrapper";
import { AssetTeaser } from "./AssetTeaser";
import { getTransformedImageVersion } from "../../../../utils/useGraphCmsImages";
var backgroundVariants = {
  tongue: 'tongue',
  upsetRect: 'upsetRect',
  rect: 'rect'
};
var assetImages = {
  none: '',
  Example1: getTransformedImageVersion('EDBzqhE5TZqIt0CEliLe'),
  Example2: getTransformedImageVersion('Qtv2VI72SrST9sPBGuuR'),
  Example3: getTransformedImageVersion('t0fyQbrNS8MT58anN0L1'),
  Example4: getTransformedImageVersion('Y6qb9VRp6F3PYl22H8gw'),
  Example5: getTransformedImageVersion('eT8kBosCTv2nQDoJdrpG'),
  Example6: getTransformedImageVersion('N1lYTvtBQo6DErnIwgpa')
};
var backgroundImages = {
  none: '',
  Example1: getTransformedImageVersion('9mSN32ckQzORsQnMC7Zc'),
  Example2: getTransformedImageVersion('SdO6Eo51RZucKUK83Um7')
};
var backgroundImagesDesktop = {
  none: '',
  Example1: getTransformedImageVersion('9Tb7RERLG05a2NplB7Sw'),
  Example2: getTransformedImageVersion('1FfvDDFERwWh35wkpNky')
};
var featuredAssetImages = [{
  image: 'public/images/featured_assets/featured_asset_danone.png'
}, {
  image: 'public/images/featured_assets/featured_asset_lxp.jpg'
}];
storiesOf('Design System/Organisms/Teasers/Asset Teaser', module).add('Default', function () {
  var knobs = function knobs() {
    return {
      smallHeadline: text('smallHeadline', 'Featured Content'),
      headline: text('Headline', 'High standards require sensible processes'),
      contentText: text('ContentText', 'Well, the answer is obvious: “He hasn’t died yet.” Fortunately, for us, Munger has kept on ticking.'),
      assetImage: radios('Asset Image', assetImages, assetImages['none']),
      backgroundImage: radios('Background Image', backgroundImages, backgroundImages['Example1']),
      backgroundImageDesktop: radios('Background Image Desktop', backgroundImagesDesktop, backgroundImagesDesktop['none']),
      backgroundVariant: radios('Background Variants', backgroundVariants, backgroundVariants['tongue']),
      featuredAssetImages: featuredAssetImages,
      badges: [{
        id: 1,
        name: 'Playlist',
        actionType: 'prominent'
      }],
      tags: array('Tags', ['73 assets', '23hours 18minutes', '3% completed'], ',')
    };
  };

  return React.createElement(StorybookRouterWrapper, null, React.createElement(AssetTeaser, _extends({}, knobs(), {
    background: "white"
  })));
});
//# sourceMappingURL=AssetTeaser.story.js.map