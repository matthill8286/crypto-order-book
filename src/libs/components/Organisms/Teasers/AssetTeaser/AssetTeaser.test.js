import React from 'react';
import { Card } from "../../../Atoms/Card";
import { mountWithThemeAndRouter, renderWithThemeAndRouter } from "../../../../testRenderer";
import { AssetTeaser } from "./AssetTeaser";
describe('TopTeaser', function () {
  it('renders correctly: default AssetTeaser with minimal configuration', function () {
    var tree = renderWithThemeAndRouter(React.createElement(AssetTeaser, {
      headline: "High standards require sensible processes",
      elevation: 0
    }));
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly: default AssetTeaser with maximum configuration', function () {
    var tree = renderWithThemeAndRouter(React.createElement(AssetTeaser, {
      backgroundImage: "public/images/small.jpg",
      elevation: 0,
      backgroundVariant: "tongue",
      contentText: "Well, the answer is obvious: \u201CHe hasn\u2019t died yet.\u201D Fortunately, for us, Munger has kept on ticking.",
      featuredAssetImages: [{
        image: 'public/images/featured_assets/featured_asset_01.png'
      }, {
        image: 'public/images/featured_assets/featured_asset_03.png'
      }],
      headline: "High standards require sensible processes",
      assetImage: "public/images/teasers/product02.png",
      tags: ['73 assets', '23hours 18minutes', '3% completed']
    }));
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly: AssetTeaser', function () {
    var tree = renderWithThemeAndRouter(React.createElement(AssetTeaser, {
      backgroundImage: "public/images/small.jpg",
      elevation: 0,
      backgroundVariant: "tongue",
      contentText: "Well, the answer is obvious: \u201CHe hasn\u2019t died yet.\u201D Fortunately, for us, Munger has kept on ticking.",
      featuredAssetImages: [{
        image: 'public/images/featured_assets/featured_asset_01.png'
      }, {
        image: 'public/images/featured_assets/featured_asset_03.png'
      }],
      headline: "High standards require sensible processes",
      tags: ['73 assets', '23hours 18minutes', '3% completed']
    }));
    expect(tree).toMatchSnapshot();
  });
  it('triggers click handler', function () {
    var clickHandler = jest.fn();
    var tree = mountWithThemeAndRouter(React.createElement(AssetTeaser, {
      elevation: 0,
      headline: "High standards require sensible processes",
      onClick: clickHandler
    }));
    tree.find(Card).simulate('click');
    expect(clickHandler).toHaveBeenCalled();
  });
});
//# sourceMappingURL=AssetTeaser.test.js.map