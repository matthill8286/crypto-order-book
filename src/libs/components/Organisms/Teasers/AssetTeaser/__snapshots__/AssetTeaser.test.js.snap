// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTeaser renders correctly: AssetTeaser 1`] = `
<a
  className="Linkstyled__StyledLinkAnchor-qsbyg3-0 kIbsuh AssetTeaserstyled__StyledLink-sc-1in05jx-3 jfwYRZ"
  color="grey4"
  target="_self"
>
  <div
    className="AssetTeaserstyled__StyledWrapper-sc-1in05jx-0 hDYMvM"
  >
    <div
      className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 kXcGQv AssetTeaserstyled__StyledCard-sc-1in05jx-1 dwdSPw"
    >
      <div
        className="Cardstyled__StyledCard-sc-1gxmhl7-1 eCZRqv"
        display="flex"
        elevation={0}
      >
        <div
          className="AssetTeaserstyled__StyledMainContainer-sc-1in05jx-4 kUthSo"
        >
          <div
            className="AssetTeaserstyled__StyledInfoContainer-sc-1in05jx-5 kysPDf"
          >
            <div
              className="AssetTeaserstyled__StyledInfoContainerContent-sc-1in05jx-6 fqkAeI"
            >
              <p
                className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 iiAVrj HeadingFeatured__StyledFeaturedHeading-sc-1i1kuqv-0 clJruv"
                color="primary"
                fontFamily="featured"
                fontSize="xxxxl"
                spacing="base"
              >
                High standards require sensible processes
              </p>
              <div
                className="AssetTeaserstyled__StyledInfoTextContainer-sc-1in05jx-7 dNbzHG"
              >
                <span
                  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 eDHZgs"
                  color="grey5"
                  fontSize="sm"
                >
                  Well, the answer is obvious: “He hasn’t died yet.” Fortunately, for us, Munger has kept on ticking.
                </span>
              </div>
              <div
                className="AssetTeaserstyled__StyledFlexBottomContainer-sc-1in05jx-14 kdBYfn"
              >
                <div
                  className="AssetTeaserstyled__StyledTopTeaserTagWrapper-sc-1in05jx-15 kEspRJ"
                >
                  <div
                    className="Tagstyled__StyledTagWrapper-k5mbhe-0 jOgjTq"
                  >
                    <span
                      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 iKQzKe"
                      color="grey3"
                      fontSize="xs"
                    >
                      73 assets
                    </span>
                  </div>
                  <div
                    className="Tagstyled__StyledTagWrapper-k5mbhe-0 jOgjTq"
                  >
                    <span
                      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 iKQzKe"
                      color="grey3"
                      fontSize="xs"
                    >
                      23hours 18minutes
                    </span>
                  </div>
                  <div
                    className="Tagstyled__StyledTagWrapper-k5mbhe-0 jOgjTq"
                  >
                    <span
                      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 iKQzKe"
                      color="grey3"
                      fontSize="xs"
                    >
                      3% completed
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="AssetTeaserstyled__StyledAssetContainer-sc-1in05jx-8 jFcHtM"
          >
            <div
              className="AssetTeaserstyled__StyledTongue-sc-1in05jx-11 kpnRRZ"
              height={140}
            >
              <div
                className="AssetTeaserstyled__StyledBackgroundImage-sc-1in05jx-10 Qfivt"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="Cardstyled__StyledFeaturedAssetsContainer-sc-1gxmhl7-3 Pjaso"
      >
        <div
          className="Cardstyled__StyledFeaturedAssetSpacing-sc-1gxmhl7-4 cUoSwD"
        >
          <div
            className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 jhkssp"
          >
            <div
              className="Cardstyled__StyledCard-sc-1gxmhl7-1 cysaNE"
              display="flex"
              elevation={1}
              shape="rounded-small"
            >
              <div
                className="Picturestyled__StyledPicture-sc-18jsup7-0 gCBPeQ Cardstyled__StyledFeaturedAssetsPicture-sc-1gxmhl7-2 gdZxmp"
                height="32px"
                width="auto"
              >
                <picture
                  style={
                    Object {
                      "height": "32px",
                      "width": "100%",
                    }
                  }
                >
                  <img
                    alt="image"
                    height="32px"
                    loading="eager"
                    src="public/images/featured_assets/featured_asset_01.png"
                    width="auto"
                  />
                </picture>
              </div>
            </div>
          </div>
        </div>
        <div
          className="Cardstyled__StyledFeaturedAssetSpacing-sc-1gxmhl7-4 cUoSwD"
        >
          <div
            className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 jhkssp"
          >
            <div
              className="Cardstyled__StyledCard-sc-1gxmhl7-1 cysaNE"
              display="flex"
              elevation={1}
              shape="rounded-small"
            >
              <div
                className="Picturestyled__StyledPicture-sc-18jsup7-0 gCBPeQ Cardstyled__StyledFeaturedAssetsPicture-sc-1gxmhl7-2 gdZxmp"
                height="32px"
                width="auto"
              >
                <picture
                  style={
                    Object {
                      "height": "32px",
                      "width": "100%",
                    }
                  }
                >
                  <img
                    alt="image"
                    height="32px"
                    loading="eager"
                    src="public/images/featured_assets/featured_asset_03.png"
                    width="auto"
                  />
                </picture>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`TopTeaser renders correctly: default AssetTeaser with maximum configuration 1`] = `
<a
  className="Linkstyled__StyledLinkAnchor-qsbyg3-0 kIbsuh AssetTeaserstyled__StyledLink-sc-1in05jx-3 jfwYRZ"
  color="grey4"
  target="_self"
>
  <div
    className="AssetTeaserstyled__StyledWrapper-sc-1in05jx-0 hDYMvM"
  >
    <div
      className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 kXcGQv AssetTeaserstyled__StyledCard-sc-1in05jx-1 dwdSPw"
    >
      <div
        className="Cardstyled__StyledCard-sc-1gxmhl7-1 eCZRqv"
        display="flex"
        elevation={0}
      >
        <div
          className="AssetTeaserstyled__StyledMainContainer-sc-1in05jx-4 kUthSo"
        >
          <div
            className="AssetTeaserstyled__StyledInfoContainer-sc-1in05jx-5 bBFFbp"
          >
            <div
              className="AssetTeaserstyled__StyledInfoContainerContent-sc-1in05jx-6 fqkAeI"
            >
              <p
                className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 iiAVrj HeadingFeatured__StyledFeaturedHeading-sc-1i1kuqv-0 clJruv"
                color="primary"
                fontFamily="featured"
                fontSize="xxxxl"
                spacing="base"
              >
                High standards require sensible processes
              </p>
              <div
                className="AssetTeaserstyled__StyledInfoTextContainer-sc-1in05jx-7 dNbzHG"
              >
                <span
                  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 eDHZgs"
                  color="grey5"
                  fontSize="sm"
                >
                  Well, the answer is obvious: “He hasn’t died yet.” Fortunately, for us, Munger has kept on ticking.
                </span>
              </div>
              <div
                className="AssetTeaserstyled__StyledFlexBottomContainer-sc-1in05jx-14 kdBYfn"
              >
                <div
                  className="AssetTeaserstyled__StyledTopTeaserTagWrapper-sc-1in05jx-15 kEspRJ"
                >
                  <div
                    className="Tagstyled__StyledTagWrapper-k5mbhe-0 jOgjTq"
                  >
                    <span
                      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 iKQzKe"
                      color="grey3"
                      fontSize="xs"
                    >
                      73 assets
                    </span>
                  </div>
                  <div
                    className="Tagstyled__StyledTagWrapper-k5mbhe-0 jOgjTq"
                  >
                    <span
                      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 iKQzKe"
                      color="grey3"
                      fontSize="xs"
                    >
                      23hours 18minutes
                    </span>
                  </div>
                  <div
                    className="Tagstyled__StyledTagWrapper-k5mbhe-0 jOgjTq"
                  >
                    <span
                      className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 iKQzKe"
                      color="grey3"
                      fontSize="xs"
                    >
                      3% completed
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="AssetTeaserstyled__StyledAssetContainer-sc-1in05jx-8 jFcHtM"
          >
            <div
              className="AssetTeaserstyled__StyledTongue-sc-1in05jx-11 kpnRRZ"
              height={140}
            >
              <div
                className="AssetTeaserstyled__StyledBackgroundImage-sc-1in05jx-10 bCJkFc"
              />
            </div>
            <div
              className="AssetTeaserstyled__StyledPictureContainer-sc-1in05jx-9 ktmXBA"
            >
              <div
                className="Picturestyled__StyledPicture-sc-18jsup7-0 jIsvqb"
                height="100%"
                width="100%"
              >
                <picture
                  style={
                    Object {
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                >
                  <img
                    alt="image"
                    height="100%"
                    loading="eager"
                    src="public/images/teasers/product02.png"
                    width="100%"
                  />
                </picture>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="Cardstyled__StyledFeaturedAssetsContainer-sc-1gxmhl7-3 Pjaso"
      >
        <div
          className="Cardstyled__StyledFeaturedAssetSpacing-sc-1gxmhl7-4 cUoSwD"
        >
          <div
            className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 jhkssp"
          >
            <div
              className="Cardstyled__StyledCard-sc-1gxmhl7-1 cysaNE"
              display="flex"
              elevation={1}
              shape="rounded-small"
            >
              <div
                className="Picturestyled__StyledPicture-sc-18jsup7-0 gCBPeQ Cardstyled__StyledFeaturedAssetsPicture-sc-1gxmhl7-2 gdZxmp"
                height="32px"
                width="auto"
              >
                <picture
                  style={
                    Object {
                      "height": "32px",
                      "width": "100%",
                    }
                  }
                >
                  <img
                    alt="image"
                    height="32px"
                    loading="eager"
                    src="public/images/featured_assets/featured_asset_01.png"
                    width="auto"
                  />
                </picture>
              </div>
            </div>
          </div>
        </div>
        <div
          className="Cardstyled__StyledFeaturedAssetSpacing-sc-1gxmhl7-4 cUoSwD"
        >
          <div
            className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 jhkssp"
          >
            <div
              className="Cardstyled__StyledCard-sc-1gxmhl7-1 cysaNE"
              display="flex"
              elevation={1}
              shape="rounded-small"
            >
              <div
                className="Picturestyled__StyledPicture-sc-18jsup7-0 gCBPeQ Cardstyled__StyledFeaturedAssetsPicture-sc-1gxmhl7-2 gdZxmp"
                height="32px"
                width="auto"
              >
                <picture
                  style={
                    Object {
                      "height": "32px",
                      "width": "100%",
                    }
                  }
                >
                  <img
                    alt="image"
                    height="32px"
                    loading="eager"
                    src="public/images/featured_assets/featured_asset_03.png"
                    width="auto"
                  />
                </picture>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`TopTeaser renders correctly: default AssetTeaser with minimal configuration 1`] = `
<a
  className="Linkstyled__StyledLinkAnchor-qsbyg3-0 kIbsuh AssetTeaserstyled__StyledLink-sc-1in05jx-3 jfwYRZ"
  color="grey4"
  target="_self"
>
  <div
    className="AssetTeaserstyled__StyledWrapper-sc-1in05jx-0 hDYMvM"
  >
    <div
      className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 kXcGQv AssetTeaserstyled__StyledCard-sc-1in05jx-1 dwdSPw"
    >
      <div
        className="Cardstyled__StyledCard-sc-1gxmhl7-1 eCZRqv"
        display="flex"
        elevation={0}
      >
        <div
          className="AssetTeaserstyled__StyledMainContainer-sc-1in05jx-4 kUthSo"
        >
          <div
            className="AssetTeaserstyled__StyledInfoContainer-sc-1in05jx-5 kysPDf"
          >
            <div
              className="AssetTeaserstyled__StyledInfoContainerContent-sc-1in05jx-6 fqkAeI"
            >
              <p
                className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 iiAVrj HeadingFeatured__StyledFeaturedHeading-sc-1i1kuqv-0 clJruv"
                color="primary"
                fontFamily="featured"
                fontSize="xxxxl"
                spacing="base"
              >
                High standards require sensible processes
              </p>
              <div
                className="AssetTeaserstyled__StyledInfoTextContainer-sc-1in05jx-7 dNbzHG"
              >
                <span
                  className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 ktxNak"
                  color="grey5"
                  fontSize="sm"
                />
              </div>
            </div>
          </div>
          <div
            className="AssetTeaserstyled__StyledAssetContainer-sc-1in05jx-8 jFcHtM"
          >
            <div
              className="AssetTeaserstyled__StyledUpsetRect-sc-1in05jx-12 bonPJA"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</a>
`;
